x/profiles/legacy/v4/store_test.go

```diff
@@ -228,7 +228,18 @@ func TestMigrateStore(t *testing.T) {
 				), stored)
 
 				// Check the application link client id
-				require.Equal(t, []byte("client_id_value"), kvStore.Get(profilestypes.ApplicationLinkClientIDKey("client_id")))
+				require.Equal(t, linkKey, kvStore.Get(profilestypes.ApplicationLinkClientIDKey("client_id")))
+			},
+		},
+		{
+			name: "leftover application client id keys are deleted properly",
+			store: func(ctx sdk.Context) {
+				kvStore := ctx.KVStore(keys[types.StoreKey])
+				kvStore.Set(v4.ApplicationLinkClientIDKey("client_id"), []byte("client_id_value"))
+			},
+			check: func(ctx sdk.Context) {
+				kvStore := ctx.KVStore(keys[types.StoreKey])
+				require.False(t, kvStore.Has(profilestypes.ApplicationLinkClientIDKey("client_id")))
```

Comment:
```suggestion
				require.False(t, kvStore.Has(v4.ApplicationLinkClientIDKey("client_id")))
```

The client id key of profilesv4 is not the same as the new one. We should check old key is deleted properly or not.

Reply:
Updated, thanks for spotting this! :pray: